chore: add build instructions to README (#3)
BugSplat-Git/bugsplat-apple: README.md

### **<div align="center">Crash and error reporting built for busy developers.</div>**

<div align="center">
    <a href="https://twitter.com/BugSplatCo">
        <img alt="Follow @bugsplatco on Bluesky" src="https://img.shields.io/badge/dynamic/json?url=https%3A%2F%2Fpublic.api.bsky.app%2Fxrpc%2Fapp.bsky.actor.getProfile%2F%3Factor%3Dbugsplatco.bsky.social&query=%24.followersCount&style=social&logo=bluesky&label=Follow%20%40bugsplatco.bsky.social">
    </a>
    <a href="https://discord.gg/bugsplat">
        <img alt="Join BugSplat on Discord" src="https://img.shields.io/discord/664965194799251487?label=Join%20Discord&logo=Discord&style=social">
    </a>
</div>

<br/>

## Introduction 👋

The BugSplat.xcframework enables posting crash reports from iOS, macOS, and Mac Catalyst applications to BugSplat. Visit [bugsplat.com](https://www.bugsplat.com) for more information and to sign up for an account.

## Requirements 📋

- BugSplat for iOS supports iOS 13 and later.
- BugSplat for macOS supports macOS 10.13 and later.

## Integration 🏗️

BugSplat supports multiple methods for installing the xcfamework in a project.

### Swift Package Manager (SPM)

> TODO BG

Author: This will be addressed in a follow up PR

> TODO BG

Author: ```suggestion
Coming soon 🙌
```

### Manual Setup

To use this xcframework in your project manually you may:

1. Download the latest release from the [Releases](https://github.com/BugSPlat-Git/bugsplat-apple/releases) page. The release will contain a zip file with the xcframework.
2. Unzip the archive.
3. In Xcode, select your app target, then go to the General tab, scroll down to Framework, Libraries, and Embedded Content, then click the "+" and navigate to locate the unzipped BugSplat.xcframework. Once added, select Embed & Sign.

## Usage 🧑‍💻

### Configuration

BugSplat requires a few Xcode configuration steps in order integrate the xcframework with your BugSplat account

Add the following case sensitive key to your app's `Info.plist` replacing `DATABASE_NAME` with your customer specific BugSplat database name.

```xml
<key>BugSplatDatabase</key>
<string>DATABASE_NAME</string>
```

> [!NOTE]
> For macOS apps, you must enable Outgoing network connections (client) in the Signing & Capabilities of the Target.

### Symbol Upload

> You must upload an archive containing your app's binary and symbols to the BugSplat server in order to symbolicate crash reports. There are scripts to help with this.

Author: ```suggestion
To symbolicate crash reports, you must upload your app's `dSYM` files to the BugSplat server. There are scripts to help with this.
```
